lib/availability_status_listener.rb

```diff
@@ -0,0 +1,78 @@
+require "manageiq-messaging"
+
+class AvailabilityStatusListener
+  SERVICE_NAME = "platform.sources.status".freeze
+  GROUP_REF = "sources-api-status-worker".freeze
+  EVENT_AVAILABILITY_STATUS = "availability_status".freeze
+
+  attr_accessor :messaging_client_options, :client
+
+  def initialize(messaging_client_options = {})
+    self.messaging_client_options = default_messaging_options.merge(messaging_client_options)
+  end
+
+  def run
+    Thread.new { subscribe_to_availability_status }
+  end
+
+  def subscribe_to_availability_status
+    Rails.logger.info("Sources API availability_status listener started...")
+
+    ManageIQ::Messaging::Client.open(messaging_client_options) do |client|
+      client.subscribe_topic(
+        :service     => SERVICE_NAME,
+        :persist_ref => GROUP_REF,
+        :max_bytes   => 500_000
+      ) do |event|
+        process_event(event)
+      end
+    end
+  rescue => e
+    Rails.logger.error(["Something is wrong with Kafka client: ", e.message, *e.backtrace].join($RS))
+    retry
```

Comment:
Does `retry` come from the `manageiq-messaging` gem?

Reply:
Retry will start another messaging client and subscribe it to the topic again.

Reply:
`retry` is a core Ruby concept: https://ruby-doc.org/docs/keywords/1.9/Object.html#method-i-retry